azure-pipelines.yml

```diff
@@ -128,167 +131,183 @@ stages:
+          namespace: $(NAMESPACE)
+
+      - task: HelmDeploy@0
+        displayName: Helm init
+        inputs:
+          connectionType: 'Kubernetes Service Connection'
+          kubernetesServiceConnection: $(CLUSTER)
+          command: init
+          arguments: --client-only
+
+      - task: HelmDeploy@0
+        displayName: Helm package
+        inputs:
+          command: package
+          chartPath: 'deployment/entity-service'
+          destination: $(Build.ArtifactStagingDirectory)
+          updatedependency: true
+
+      - task: KubernetesManifest@0
+        name: bake
+        displayName: Bake K8s manifests from Helm chart
+        inputs:
+          action: 'bake'
+          helmChart: 'deployment/entity-service'
+          releaseName: $(DEPLOYMENT)
+          overrides: |
+            api.app.debug:true
+            global.postgresql.postgresqlPassword:notaproductionpassword
+            api.ingress.enabled:false
+            workers.replicaCount:4
+            api.www.image.repository:$(frontendImageName)
+            api.app.image.repository:$(backendImageName)
+            api.dbinit.image.repository:$(backendImageName)
+            workers.image.repository:$(backendImageName)
+
+      - task: KubernetesManifest@0
+        displayName: Deploy K8s manifest
+        inputs:
+          action: 'deploy'
+          kubernetesServiceConnection: '$(CLUSTER)'
+          manifests: '$(bake.manifestsBundle)'
+          namespace: $(NAMESPACE)
+          containers: |
+            data61/anonlink-nginx:$(DOCKER_TAG)
+            data61/anonlink-app:$(DOCKER_TAG)
+            #data61/anonlink-worker:$(DOCKER_TAG)
+
+      # The file `.azurePipeline/k8s_test_job.yaml.tmpl` is a template for the creation of the test job.
+      # The test job stops when all the tests have been run, creating a test result file in the PVC.
+      - script: |
+          cat .azurePipeline/k8s_test_job.yaml.tmpl | \
+          sed 's|\$PVC'"|$(PVC)|g" | \
+          sed 's|\$DEPLOYMENT_NAME'"|$(DEPLOYMENT)|g" | \
+          sed 's|\$IMAGE_NAME_WITH_TAG'"|$(IMAGE_NAME_WITH_TAG)|g" | \
+          sed 's|\$SERVICE'"|$(SERVICE)|g" > $(Build.ArtifactStagingDirectory)/k8s_test_job.yaml
+        displayName: 'Prepare integration test job from template'
+
+      - task: KubernetesManifest@0
+        displayName: 'Deploy Integration Test Job'
+        inputs:
+          action: 'deploy'
+          kubernetesServiceConnection: '$(CLUSTER)'
+          manifests: '$(Build.ArtifactStagingDirectory)/k8s_test_job.yaml'
+          namespace: $(NAMESPACE)
+
+      - task: Kubernetes@1
+        displayName: 'Wait for integration test pod to be ready'
+        name: waitpods
+        inputs:
+          connectionType: 'Kubernetes Service Connection'
+          kubernetesServiceEndpoint: $(CLUSTER)
+          command: 'wait'
+          arguments: '--namespace $(NAMESPACE) -ldeployment=$(DEPLOYMENT) --timeout=30s --for=condition=Ready pods'
+
+      - task: Kubernetes@1
```

Comment:
Just wondering what is the difference between Kubernetes@0 and Kubternetes@1?

Reply:
Versions of the task - if there is a `Kubernetes@0` I should update it

Reply:
Looks like they are fine. If you're interested in the docs have a look [here](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/kubernetes?view=azure-devops)